### ISSUE TITLE
Add frontend edit capability for particles

### ISSUE THREAD

**Issue by marktaylor46** [gantry/gantry5#1005]
This follows on from #966 and member requests in the forum.
@mahagr asked me to raise this ticket. This is a feature request.

We need to provide the ability to edit particles in the frontend. As an example, in Joomla you can edit modules and articles in the frontend. It seem logical that we should also provide the ability to edit particles too to provide complete "content" edit ability. Why? because a lot of developers develop a site on behalf of their client and then let them edit there content thereafter via the frontend (not backend).

**Comment by rsroman** [comment]
Is something being done about this request? I've just begun using Gantry 5, and realized this issue. It will be absolutely critical that front end users on the sites I design will be able to update particles without have to navigate the back end. When will this be implemented?

**Comment by mahagr** [comment]
We have talked about implementing this feature, but it hasn't been assigned to a release yet.

**Comment by lauroguedes** [comment]
It is release will be great, because it will able the user to do edit way a lot simple the layout.

**Comment by lucylyn** [comment]
Any update on when this might be released, I love particles but if clients cannot make changes on frontend then this is a problem.

**Comment by lauroguedes** [comment]
@lucylyn I agree with you

**Comment by sedonasky** [comment]
With Gantry 4 I could train a client how to manage their site. Not any more...

Particles is the worst idea in programming history. Seriously. Why try to out-think the beauty of Joomla modules?

Adding a front-end editor will then make Particles the second worst idea (after Gantry 5).

**Comment by w00fz** [comment]
Didn't you hear? Gantry is not for Joomla only... Particles give developers a much easier way to develop "modules/widgets" for the site, all you need to know is basic HTML/Twig and CSS. Have you ever tried creating a module with Joomla for simple things?

Did you see the admin of Gantry 5? Can't you train your clients to edit particles in there?

**Comment by 3cellhosting** [comment]
The advantages of Gantry 5 are subjective. We have members who need complex layouts that simply could not be achieved with Gantry 4. Other members want to build their own particles so customers cannot edit them or break them. We also have members who want to build quick and easy websites, either for clients or themselves. In this respect they can use Gantry 5 and set the outlines to just contain module positions and no particles. Then they can use any modules they wish, with front end editing, in just the same way as Gantry 4 but using Gantry 5 as a faster framework.

**Comment by JoomFX** [comment]
@sedonasky Hahaha, you are joking, right? :smile: 

> Particles is the worst idea in programming history. Seriously. Why try to out-think the beauty of Joomla modules?

No, no mate... the Particle system in Gantry 5 is one of the most awesome things that have happened in the web development industry in the recent years.
You can build a particle in 15mins, literally. And you can fetch Joomla content in the particles directly, so you do not need to use any 3rd party extensions.
In order to build a Joomla module that fetched articles automatically, you would need several days (if not weeks) and a quite good knowledge in PHP/SQL. While you can do that for 15mins with the G5 Particles. And then, guess what... you can use the "Gantry 5 Particle" module (in the Joomla Module Manager) to publish it as you would do with any other Joomla Module.

> Adding a front-end editor will then make Particles the second worst idea (after Gantry 5).

Just a quick question for you, but please be honest - why do you think that "the worst idea" (according to your comment above) has been voted as [the best Joomla Framework at "J and Beyond" in Barcelona](http://jandbeyond.org/news/571-j-o-s-c-a-r-s-winners-2016.html)?

**Comment by StefanvDelft** [comment]
As I am a beginning website builder, I'm quite new in the world of template frameworks. I have some experience with Helix 3 and right now I'm exploring the Gantry 5 framework. At first sight, it's very intuitive and easy to use and I'm happy that I chose to work with it.

For the frontend editing feature, I hope that it will be implemented in the near future.
For me it's a nice-to-have: because of the user friendly Gantry 5 framework I will definitely be able to train my clients to edit the particles in the backend. But for the lazy clients who would like to just click on the frontend and edit, this feature would be really nice.

**Comment by shouldbediving** [comment]
Any updates on this? or might there be a workaround at least for the article particle?

**Comment by mahagr** [comment]
No updates yet. We are just bug fixing phase in G5.4 and after it has been released, this feature will be looked at.

**Comment by N8Solutions** [comment]
Just noticed this issue yesterday when I was about to walk a client through on how to do some front end editing on the website. I look forward to this being implemented.

**Comment by twdjo** [comment]
Plus 1 for this issue. Would be great to not have to give back-end access to a user just to make changes to home page content modules created with Gantry 5.

**Comment by lucylyn** [comment]
As I have said earlier frontend access would be great.  In the meantime I have been using some particles from Inspire Theme  [http://www.inspiretheme.com/gantry5-particles](url) and they have a few that pull content from Joomla articles which can work well for clients.

**Comment by twdjo** [comment]
@lucylyn Thanks. I guess I misspoke when I used the term "content". I meant any text or images that are on the home page and generated from Gantry 5 particles, not just Joomla articles.

**Comment by omikron** [comment]
I know this is still being looked at, and I don't know how feasible this would be, but if the front end edit icon could directly open the particles modal window - that would be amazing for usability.

**Comment by vastozone** [comment]
I'm building a government website using Inspiretheme's Oxygen tempalte - Gantry 5. Am I correct in saying that there is NO way to edit the frontend by my staff using the Mouse-over Edit Icons option in the Global Configurations?

**Comment by mahagr** [comment]
Frontend editing works for everything else but the particles in the page.

**Comment by vastozone** [comment]
May I ask how to enable that function. I have enabled Mouse-over Edit Icons for: Modules & Menus. 
What needs to be done to allow editing?
Thank you.

**Comment by mahagr** [comment]
For frontend editing, Gantry needs support for accessing parts of Gantry admin from the frontend, including support for AJAX.

**Comment by N8Solutions** [comment]
Just checking in on this to see if any progress has been made.

Is it possible we'll see this feature sometime soon?

**Comment by xoomserve** [comment]
Me too, Just checking in on this to see if any progress has been made. Any news?

Thank you

**Comment by mahagr** [comment]
I just added some AJAX support to the frontend, so some progress has been done towards this one.

**Comment by ShibumiDesign** [comment]
Definitely needs the ability to edit particles in the front end, especially particles like "Info List", "Block Content", "Content List" and others, more specifically, particles that contain simple text information. I totally agree with many of the other user comments that having to give end users back end access to edit particles is not the way to do this - I actually had one client who had backend access inadvertantly revert their joomla installation back to a fresh install losing everything because they had access to the backend and didn't know what they were doing. End users with backend access usually equals trouble waiting to happen.

Now I know this could potentially end up being a double edged sword, in that the more access to things we give end users in the front end, could lead to other problems, but if we can already edit modules and articles in the front end, why shouldn't we be able to edit a majority of particles in the front end?? If a user is logged in and has edit capabilities. then have an edit particle icon or text that allows us to edit the particle from the front end...

**Comment by muffindcc** [comment]
Any news on implementing this?

**Comment by topwebs** [comment]
Would just enabling front-end editing for Gantry 5 modules (with particles) be easier than front-end editing of particles?

**Comment by marktaylor46** [comment]
I don't think it would be but in any case we need one generic solution for Gantry 5 across all three of the CMSs it supports.

But I am sure @mahagr will comment further.

**Comment by mahagr** [comment]
There's no difference really, they still need to be handled in the same way.

**Comment by Spilloljan** [comment]
Anything happening with Gantry 5 so that "Inline Editing" will work in Frontend?

**Comment by mahagr** [comment]
Right now frontend editing isn't of high priority. We will revisit our plans once higher priority tasks such as Joomla 4 support is implemented.

**Comment by 1apweb** [comment]
Hi, It's really important to be able to modify the modules on the frontend. I create sites for beginners and this would allow them to update their site quickly. There is a great demand and it would really be better for Gantry.

**Comment by mediaDESIGN-SK** [comment]
I have done a lot of customer projects with Gantry5 and I love the fast and effective creation of particles. I programmed a lot of my own particles and think it's fantastic that I can use them in Wordpress and Gravity projects without changing them.

But currently I can use Gantry5 less and less because the customers want to edit posts in the frontend and that's exactly what Particles is currently not able to do. That is a shame and could mean that I have to turn my back on Gantry. I don't want to support several template frameworks but rather concentrate on one tool. Please enable front editing of Gantry5 particles - urgently!

I don't understand why one can doubt the urgency and necessity.

**Comment by mediaDESIGN-SK** [comment]
Of course, Joomla4 support is absolutely urgent. But also in Joomla4, front-end editing is becoming more and more important and better. A main argument why many project customers prefer Wordpress in the end is that frontend editing is much more advanced there. Joomla4 has to support it solidly and in full and thus also Gantry5. I also don't understand why it takes so many years to enable the great Particle form in the frontend.
